lib/cloudant.js

```diff
@@ -540,11 +540,59 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
       return cb();
     }, function(err) {
       if (err) return cb(err);
-      mo.db.bulk({docs: docs}, function(err) {
+      mo.db.bulk({docs: docs}, function(err, result) {
         if (err) return cb(err);
-        var res = {};
-        res.count = docs.length;
-        return cb(err, res);
+        var errorArray = _.filter(result, 'error');
+        if (errorArray.length > 0) {
+          err = new Error(g.f(util.format('Unable to update 1 or more ' +
+          'document(s): %s', util.inspect(errorArray, 2))));
+          return cb(err);
+        } else {
+          return cb(err, {count: result.length});
+        }
+      });
+    });
+  });
+};
+
+/**
+ * Perform a bulk update on a model instance
+ *
+ * @param {String} model The model name
+ * @param {Array} datas List of data to be updated
+ * @callback {Function} cb The callback function
+ */
+Cloudant.prototype.massUpdate = function(model, datas, cb) {
+  debug('Cloudant.prototype.massUpdate %j %j',
+          datas);
+  var self = this;
+  var mo = self.selectModel(model);
+
+  var toBeUpdatedList = [];
+  var data = {};
+
+  self.all(model, {where: {}}, {raw: true}, function(err, docs) {
```

Comment:
Unlike the update/updateAll function, we don't want to call `self.all`, since we're just receiving an array of objects to replace in Cloudant.